<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from cobod.urdf.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="cobod_arm">
  <!-- <xacro:property name="op_mode" default="${int(arg('op_mode'))}"/> -->
  <!-- create link fixed to the "world" -->
  <link name="world"/>
  <ros2_control name="cobod_arm" type="system">
    <hardware>
      <plugin>mock_components/GenericSystem</plugin>
      <param name="mock_sensor_commands">False</param>
      <param name="state_following_offset">0.0</param>
    </hardware>
    <joint name="Joint_0">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <command_interface name="effort"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="effort">
        <param name="initial_value">0.0</param>
      </state_interface>
    </joint>
    <joint name="Joint_1">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <command_interface name="effort"/>
      <state_interface name="position">
        <param name="initial_value">0.785</param>
      </state_interface>
      <state_interface name="velocity">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="effort">
        <param name="initial_value">0.0</param>
      </state_interface>
    </joint>
    <joint name="Joint_2">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <command_interface name="effort"/>
      <state_interface name="position">
        <param name="initial_value">-0.785</param>
      </state_interface>
      <state_interface name="velocity">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="effort">
        <param name="initial_value">0.0</param>
      </state_interface>
    </joint>
    <joint name="Joint_3">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <command_interface name="effort"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="effort">
        <param name="initial_value">0.0</param>
      </state_interface>
    </joint>
    <joint name="Joint_4">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <command_interface name="effort"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="effort">
        <param name="initial_value">0.0</param>
      </state_interface>
    </joint>
    <joint name="Joint_5">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <command_interface name="effort"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="effort">
        <param name="initial_value">0.0</param>
      </state_interface>
    </joint>
  </ros2_control>
  <link name="base_link"/>
  <link name="base_link_inertia">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cobod_description/meshes/cobod_arm/visual/base_link.STL" scale="1 1 1"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cobod_description/meshes/cobod_arm/collision/base_link.STL" scale="1 1 1"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.0021349 -0.0073177 0.21837"/>
      <mass value="17.658"/>
      <inertia ixx="0.49696" ixy="0.090037" ixz="7.1468E-08" iyy="0.42557" iyz="1.5488E-06" izz="0.69453"/>
    </inertial>
  </link>
  <link name="Link_1">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cobod_description/meshes/cobod_arm/visual/Link_1.STL" scale="1 1 1"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cobod_description/meshes/cobod_arm/collision/Link_1.STL" scale="1 1 1"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-5.5969E-05 -0.025598 0.41214"/>
      <mass value="39.676"/>
      <inertia ixx="0.5089" ixy="5.8444E-06" ixz="-4.8303E-05" iyy="0.60162" iyz="0.00044952" izz="0.62118"/>
    </inertial>
  </link>
  <link name="Link_2">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cobod_description/meshes/cobod_arm/visual/Link_2.STL" scale="1 1 1"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cobod_description/meshes/cobod_arm/collision/Link_2.STL" scale="1 1 1"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.36598 -0.0066396 0.069396"/>
      <mass value="24.392"/>
      <inertia ixx="0.81184" ixy="-0.0020425" ixz="0.018017" iyy="0.59313" iyz="-0.00384" izz="0.903"/>
    </inertial>
  </link>
  <link name="Link_3">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cobod_description/meshes/cobod_arm/visual/Link_3.STL" scale="1 1 1"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cobod_description/meshes/cobod_arm/collision/Link_3.STL" scale="1 1 1"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.090213 -0.0077131 0.17487"/>
      <mass value="36.463"/>
      <inertia ixx="0.49076" ixy="-0.00078713" ixz="-0.060046" iyy="0.47838" iyz="0.0039318" izz="0.67954"/>
    </inertial>
  </link>
  <link name="Link_4">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cobod_description/meshes/cobod_arm/visual/Link_4.STL" scale="1 1 1"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cobod_description/meshes/cobod_arm/collision/Link_4.STL" scale="1 1 1"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.26721 -0.012369 0.0026916"/>
      <mass value="14.394"/>
      <inertia ixx="0.077952" ixy="8.9234E-06" ixz="-5.9892E-05" iyy="0.066882" iyz="-2.7942E-06" izz="0.089284"/>
    </inertial>
  </link>
  <link name="Link_5">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cobod_description/meshes/cobod_arm/visual/Link_5.STL" scale="1 1 1"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cobod_description/meshes/cobod_arm/collision/Link_5.STL" scale="1 1 1"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.17268 -0.010021 0.011228"/>
      <mass value="8.3473"/>
      <inertia ixx="0.13104" ixy="0.0016686" ixz="0.0015713" iyy="0.11285" iyz="-0.00080477" izz="0.1005"/>
    </inertial>
  </link>
  <link name="Gripper_base">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cobod_description/meshes/cobod_arm/visual/Gripper_base.STL" scale="1 1 1"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cobod_description/meshes/cobod_arm/collision/Gripper_base.STL" scale="1 1 1"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.019514 -0.013312 -0.00030703"/>
      <mass value="6.948"/>
      <inertia ixx="0.026728" ixy="-1.9455E-05" ixz="-4.6925E-05" iyy="0.025662" iyz="2.9951E-07" izz="0.023626"/>
    </inertial>
  </link>
  <joint name="base_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="world"/>
    <child link="base_link"/>
  </joint>
  <joint name="base_link-base_link_inertia" type="fixed">
    <parent link="base_link"/>
    <child link="base_link_inertia"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <joint name="Joint_0" type="revolute">
    <parent link="base_link_inertia"/>
    <child link="Link_1"/>
    <origin rpy="0 0 0" xyz="0 0 0.3072"/>
    <axis xyz="0 0 1"/>
    <limit effort="40" lower="-3.141592653589793" upper="3.141592653589793" velocity="0.0"/>
    <dynamics damping="0" friction="0"/>
  </joint>
  <joint name="Joint_1" type="revolute">
    <parent link="Link_1"/>
    <child link="Link_2"/>
    <origin rpy="0 -0.197 0" xyz="0 0 0.529"/>
    <axis xyz="0 1 0"/>
    <limit effort="40" lower="0.0" upper="2.2689280275926285" velocity="3.141592653589793"/>
    <dynamics damping="0" friction="0"/>
  </joint>
  <joint name="Joint_2" type="revolute">
    <parent link="Link_2"/>
    <child link="Link_3"/>
    <origin rpy="0 0.385 0" xyz="-1.0267 0 0.19465"/>
    <axis xyz="0 1 0"/>
    <limit effort="40" lower="-3.141592653589793" upper="-0.12217304763960307" velocity="3.141592653589793"/>
    <dynamics damping="0" friction="0"/>
  </joint>
  <joint name="Joint_3" type="revolute">
    <parent link="Link_3"/>
    <child link="Link_4"/>
    <origin rpy="0 -0.19 0" xyz="0.66415 0.015933 0.41691"/>
    <axis xyz="1 0 0"/>
    <limit effort="40" lower="-1.5707963267948966" upper="1.5707963267948966" velocity="3.141592653589793"/>
    <dynamics damping="0" friction="0"/>
  </joint>
  <joint name="Joint_4" type="revolute">
    <parent link="Link_4"/>
    <child link="Link_5"/>
    <origin rpy="0 0 0" xyz="0.33966 -0.015933 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="50" lower="-1.5707963267948966" upper="1.5707963267948966" velocity="3.141592653589793"/>
    <dynamics damping="0" friction="0"/>
  </joint>
  <joint name="Joint_5" type="revolute">
    <parent link="Link_5"/>
    <child link="Gripper_base"/>
    <origin rpy="0 0 0" xyz="0.385 -0.0018286 0.006823"/>
    <axis xyz="0 1 0"/>
    <limit effort="50" lower="-1.5707963267948966" upper="1.5707963267948966" velocity="3.141592653589793"/>
    <dynamics damping="0" friction="0"/>
  </joint>
  <link name="flange"/>
  <joint name="Gripper_base-flange" type="fixed">
    <parent link="Gripper_base"/>
    <child link="flange"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <link name="tool0"/>
  <joint name="flange-tool0" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="flange"/>
    <child link="tool0"/>
  </joint>
</robot>
